modules/core/src/transitions/gpu-interpolation-transition.js

```diff
@@ -50,47 +47,49 @@ export default class GPUInterpolationTransition {
   // in case the attribute's buffer has changed in length or in
   // bufferLayout
   start(transitionSettings, numInstances) {
-    assert(
-      transitionSettings.duration > 0,
-      'transition setting must have a duration greater than 0'
-    );
+    if (transitionSettings.duration <= 0) {
+      this.transition.cancel();
+      return;
+    }
+
+    const {gl, buffers, attribute} = this;
     // Alternate between two buffers when new transitions start.
     // Last destination buffer is used as an attribute (from state),
     // And the other buffer is now the current buffer.
-    cycleBuffers(this.buffers);
+    cycleBuffers(buffers);
 
     const padBufferOpts = {
       numInstances,
-      attribute: this.attribute,
+      attribute,
       fromLength: this.currentLength,
       fromBufferLayout: this.currentBufferLayout,
       getData: transitionSettings.enter
     };
 
-    for (const buffer of this.buffers) {
+    for (const buffer of buffers) {
       padBuffer({buffer, ...padBufferOpts});
     }
 
-    this.currentBufferLayout = this.attribute.bufferLayout;
-    this.currentLength = getAttributeBufferLength(this.attribute, numInstances);
+    this.currentBufferLayout = attribute.bufferLayout;
+    this.currentLength = getAttributeBufferLength(attribute, numInstances);
     this.attributeInTransition.update({
-      buffer: this.buffers[1],
+      buffer: buffers[1],
       // Hack: Float64Array is required for double-precision attributes
       // to generate correct shader attributes
-      value: this.attribute.value
+      value: attribute.value
     });
 
     this.transition.start(transitionSettings);
 
-    this.transform = this.transform || new Transform(this.gl, getShaders(this.attribute.size));
+    this.transform = this.transform || new Transform(gl, getShaders(attribute.size));
```

Comment:
Because we are dropping out of this function at the top in some cases, `this.transform` doesn't always get created (since we're lazily creating it here). Can we move the `this.transform` creation to the `constructor` with the other GL resources? So we can easily reason about the state of this instance at any given point in time?

Reply:
@Pessimistress - I moved `this.transform` creation to `constructor` in the commit I just pushed.